tonic-web/src/call.rs

```diff
@@ -237,49 +251,61 @@ where
     type Data = Bytes;
     type Error = Status;
 
-    fn poll_data(
+    fn poll_frame(
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-    ) -> Poll<Option<Result<Self::Data, Self::Error>>> {
+    ) -> Poll<Option<Result<http_body::Frame<Self::Data>, Self::Error>>> {
         if self.client && self.direction == Direction::Decode {
             let mut me = self.as_mut();
 
             loop {
-                let incoming_buf = match ready!(me.as_mut().poll_decode(cx)) {
-                    Some(Ok(incoming_buf)) => incoming_buf,
-                    None => {
-                        // TODO: Consider eofing here?
```

Comment:
I think so - because we can continue polling the decoder waiting for trailers, and we can't switch to calling `poll_trailers`, so we no longer bail when there is no more data.